app/registries/foreman/plugin/report_scanner_registry.rb

```diff
@@ -0,0 +1,30 @@
+require_dependency File.expand_path('../../../../services/report_scanner/puppet_report_scanner', __FILE__)
+
+module Foreman
+  class Plugin
+    class ReportScannerRegistry
+      DEFAULT_REPORT_SCANNER = [
+        ::Foreman::PuppetReportScanner
+      ].freeze
+
+      attr_accessor :report_scanner
+
+      def initialize
+        @report_scanner = []
+        register_report_scanner DEFAULT_REPORT_SCANNER
+      end
+
+      def report_scanner
+        @report_scanner ||= []
+      end
+
+      def register_report_scanner(scanner)
+        @report_scanner = (report_scanner << scanner).flatten.uniq
```

Comment:
why `flatten` is required here?

Reply:
:man_facepalming: because I called it with an array to register the default scanner all this time...